packages/client/src/provider/in-memory-provider/in-memory-provider.ts

```diff
@@ -0,0 +1,186 @@
+    try {
+
+      for (const key in this._flagConfiguration) {
+        if (this._flagConfiguration[key].disabled) {
+          continue;
+        }
+
+        this._flagEvalCache[key] = this.resolveFlagWithReason(key, context);
+
+      }
+
+      this.events.emit(ProviderEvents.Ready);
+    } catch (error) {
+      this.status = ProviderStatus.ERROR;
+      this.events.emit(ProviderEvents.Error);
+      throw error;
+    }
+  }
+
+  /**
+   * Overwrites the configured flags.
+   * @param { FlagConfiguration } flagConfiguration new flag configuration
+   */
+  putConfiguration(flagConfiguration: FlagConfiguration) {
+    const flagsChanged = Object.entries(flagConfiguration)
+      .filter(([key, value]) => this._flagConfiguration[key] !== value)
+      .map(([key]) => key);
+
+    this.status = ProviderStatus.STALE;
+    this.events.emit(ProviderEvents.Stale);
+
+    this._flagConfiguration = { ...flagConfiguration };
+    this.events.emit(ProviderEvents.ConfigurationChanged, { flagsChanged });
+  }
+
+  /**
+   * Overwrites the configured flags & updates internal cache (resolved flags).
+   * @param { FlagConfiguration } flagConfiguration new flag configuration
+   * @param { EvaluationContext } [context] flag evaluation context
+   */
+  async putConfigurationAndUpdateResolution(flagConfiguration: FlagConfiguration, context?: EvaluationContext | undefined): Promise<void> {
+    this.putConfiguration(flagConfiguration);
+
+    await this.initialize(context);
+  }
+
+  resolveBooleanEvaluation(
+    flagKey: string,
+    defaultValue: boolean,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): ResolutionDetails<boolean> {
+    return this.resolveFromCache<boolean>(flagKey, defaultValue, logger);
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): ResolutionDetails<number> {
+    return this.resolveFromCache<number>(flagKey, defaultValue, logger);
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): ResolutionDetails<string> {
+    return this.resolveFromCache<string>(flagKey, defaultValue, logger);
+  }
+
+  resolveObjectEvaluation<T extends JsonValue>(
+    flagKey: string,
+    defaultValue: T,
+    context?: EvaluationContext,
+    logger?: Logger,
+  ): ResolutionDetails<T> {
+    return this.resolveFromCache<T>(flagKey, defaultValue, logger);
+  }
+
+  private resolveFromCache<T extends JsonValue | FlagValueType>(flagKey: string,
+    defaultValue: T, logger?: Logger): ResolutionDetails<T> {
+    if (!(flagKey in this._flagEvalCache)) {
+      const message = `no flag found with key ${flagKey}`;
+      logger?.debug(message);
+      throw new FlagNotFoundError(message);
+    }
+
+    if (this._flagConfiguration[flagKey].disabled) {
+      return { value: defaultValue, reason: StandardResolutionReasons.DISABLED };
+    }
```

Comment:
This is somewhat odd. As a user of this provider, I do not know of the difference between cache and flagConfiguration.
Here, the configuration might change the resolution, even though the cache might not be disabled.
My feeling is that we should only have one source of truth for resolving.
If we want to keep the cache, then we should use the `caches disabled value` of the flag here.